addons/mail/static/tests/suggestion/suggestion_tests.js

```diff
@@ -215,3 +215,72 @@ QUnit.test("Suggestions are shown after delimiter was used in text (#)", async (
     await insertText(".o-mail-Composer-input", "#");
     assert.containsOnce($, ".o-mail-Composer-suggestion:contains(General)");
 });
+
+QUnit.test("Internal user should be displayed first", async (assert) => {
+    const pyEnv = await startServer();
+    const userId = pyEnv["res.users"].create({ share: true });
+    const partnerIds = pyEnv["res.partner"].create([
+        {
+            email: "partner@example.com",
+            name: "Trigger A simple partner",
+        },
+        {
+            email: "follower@example.com",
+            name: "Trigger B follower",
+        },
+        {
+            email: "Internal@example.com",
+            name: "Trigger C internal user",
+            user_ids: [userId],
+        },
+        {
+            email: "Internal+Follower@example.com",
+            name: "Trigger D internal + follower",
+            user_ids: [userId],
+        },
+        {
+            email: "partner@example.com",
+            name: "Trigger Y simple partner",
+        },
+        {
+            email: "partner@example.com",
+            name: "Trigger X simple partner",
+        },
+        {
+            email: "apartner@example.com",
+            name: "Trigger X simple partner",
+        },
+    ]);
```

Comment:
Keep name short, it's hard to read test when the name are sentences.
`A B C D` or `John James Jimmy Demo`, as long as this is short and easy to read